src/Microsoft.AspNetCore.Authentication.JwtBearer/JwtBearerHandler.cs

```diff
@@ -185,11 +185,20 @@ internal class JwtBearerHandler : AuthenticationHandler<JwtBearerOptions>
 
         protected override async Task<bool> HandleUnauthorizedAsync(ChallengeContext context)
         {
-            var authResult = await HandleAuthenticateOnceAsync();
+            Exception failure = null;
```

Comment:
As you mention already we don't necessarily know what happens in HandleAuthenticate since its virtual. In general we only call the Once method I believe, so I think it's fine to just normalize our calls there as a result. Effectively the Once method can be thought of as EnsureNormalizedAuthResult (we could introduce a new method that wraps Once instead of changing the behavior but that will require more code changes for any middleware to pick up the new behavior). 

We don't know what people who are calling HandleAuthenticate explicitly are doing today since that's not what we do, so I don't think we can or should try to change the behavior of that virtual method, only the wrapper.